pkg/netpol/eval/check.go

```diff
@@ -240,11 +240,17 @@ func (pe *PolicyEngine) allallowedXgressConnections(src, dst k8s.Peer, isIngress
 			return common.MakeConnectionSet(true), nil // all connections allowed - no restrictions on ingress to externalIP
 		}
 		netpols, err = pe.getPoliciesSelectingPod(dst.GetPeerPod(), netv1.PolicyTypeIngress)
+		if len(netpols) > 0 {
+			dst.GetPeerPod().IngressProtected = true
```

Comment:
done in [commit](https://github.com/np-guard/netpol-analyzer/pull/296/commits/63e8f68baff1dfb0434fd69de5bcdd9a680a8fd9) ;
since the func gets the direction, moved the new func to `getPoliciesSelectingPod`, so it will have one call and be updated definitely for the pod